cmd/export.go

```diff
@@ -0,0 +1,164 @@
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		var out string
+		var format string
+
+		out, err = cmd.Flags().GetString("out")
+
+		if err != nil {
+			return fmt.Errorf("could not get out flag: %w", err)
+		}
+
+		format, err = cmd.Flags().GetString("format")
+
+		if err != nil {
+			return fmt.Errorf("could not get format flag: %w", err)
+		}
+
+		if out == "" {
+			out = "STD_OUT"
+		}
+
+		if format == "" {
+			return fmt.Errorf("format is required")
+		}
+
+		if format != "dotenv" && format != "json" && format != "yaml" {
+			return fmt.Errorf("invalid format %s", format)
+		}
+
+		storage := viper.GetString(configKeyStorage)
+
+		b, err := backend.Backends[storage].NewContext(cmdContext, viper.AllSettings())
+		if err != nil {
+			return err
+		}
+
+		exists, err := b.ExistsContext(cmdContext)
+		if err != nil {
+			return fmt.Errorf("could not check store existence: %w", err)
+		}
+		if !exists {
+			return fmt.Errorf("store does not exist")
+		}
+
+		data, err := b.LoadContext(cmdContext)
+		if err != nil {
+			return fmt.Errorf("could not load data from store: %w", err)
+		}
+
+		password := []byte(viper.GetString(configKeyPassword))
+		s, err := store.ReadStoreContext(cmdContext, password, data)
+		if err != nil {
+			return fmt.Errorf("could not read store from data: %w", err)
+		}
+
+		keys := s.ListContext(cmdContext)
+
+		var sb strings.Builder
+
+		if format == "json" {
+			sb.WriteString("{\n")
+
+			for i, key := range keys {
+				if i > 0 {
+					sb.WriteString(",\n")
+				}
+
+				sb.WriteString(`"` + key + `": `)
+
+				val, err := s.GetContext(cmdContext, key)
+
+				if err != nil {
+					return fmt.Errorf("could not get value for key %s: %w", key, err)
+				}
+
+				sb.WriteString(`"` + string(val) + `"`)
+			}
+
+			sb.WriteString("\n}")
+		} else if format == "yaml" {
+
+			sb.WriteString("---\n")
+
+			for _, key := range keys {
+				sb.WriteString(key)
+				sb.WriteString(": ")
+
+				val, err := s.GetContext(cmdContext, key)
+
+				if err != nil {
+					return fmt.Errorf("could not get value for key %s: %w", key, err)
+				}
+
+				sb.WriteString(string(val))
+				sb.WriteString("\n")
+			}
+
+		} else if format == "dotenv" {
+			for _, key := range keys {
+				sb.WriteString(key)
+				sb.WriteString("=")
+
+				val, err := s.GetContext(cmdContext, key)
+
+				if err != nil {
+					return fmt.Errorf("could not get value for key %s: %w", key, err)
+				}
+
+				sb.WriteString(string(val))
+				sb.WriteString("\n")
+			}
+		}
+
+		if out == "STD_OUT" {
+			fmt.Println(sb.String())
+			return nil
+		}
+
+		err = os.WriteFile(out, []byte(sb.String()), 0644)
+
+		if err != nil {
+			return fmt.Errorf("could not write file %s: %w", out, err)
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	exportCmd.Flags().StringP("out", "o", "STD_OUT", "export to file (defaults to stdout)")
+
+	exportCmd.Flags().StringP("format", "f", "", "export file format (json,yaml,dotenv)")
+
+}
```

Comment:
This is more of an open question, but what do you think we let export flags be configured with the config file?

If our user plans on using `scrt` regularly, it could be worth having a config like:

```yaml
output: ".env"
format: dotenv
```

And then just running

```shell
scrt export
```

to re-generate the file.

This would mean binding the pflags to viper, as done [here](https://github.com/loderunner/scrt/blob/8f36e7843e02009994510522be691d73685c4c19/cmd/root.go#L179-L180).

Not a mandatory feature, just let me know what you think.

Reply:
I think this is a great idea! I'm going to resolve your requested changes and will circle back

Reply:
Done!